diff --git a/src/client/mod.rs b/src/client/mod.rs
index 4fff27f..8df524d 100644
--- a/src/client/mod.rs
+++ b/src/client/mod.rs
@@ -325,6 +325,13 @@ impl ClientBuilder {
         self
     }
 
+    /// Sets an event handler with multiple methods for each possible event. Uses an Arc
+    pub fn event_handler_arc<H: EventHandler + 'static>(mut self, event_handler: Arc<H>) -> Self {
+        self.event_handler = Some(event_handler);
+
+        self
+    }
+
     /// Sets an event handler with a single method where all received gateway
     /// events will be dispatched.
     pub fn raw_event_handler<H: RawEventHandler + 'static>(mut self, raw_event_handler: H) -> Self {
